crates/turbopack-ecmascript/src/tree_shake/mod.rs

```diff
@@ -322,16 +355,60 @@ async fn get_part_id(result: &SplitResult, part: Vc<ModulePart>) -> Result<u32>
         }
     };
 
-    let entrypoints = match &result {
-        SplitResult::Ok { entrypoints, .. } => entrypoints,
-        _ => bail!("split failed"),
+    let SplitResult::Ok {
+        entrypoints,
+        modules,
+        ..
+    } = &result
+    else {
+        bail!("split failed")
     };
```

Comment:
Outside of the context of this PR, but why do we implement our own enum type here instead of using something like `Result<Split, Vc<ParseResult>>`? If this used a more standard `Result` we could've used `.context()`.

Reply:
We can't use `.context()` on `Result<T, E>` anyway if `E` is `Vc< ParseResult>`. `.context()` wants `E` to implement specific traits, and `Vc<ParseResult>` does not fit for `Result::Err` IMHO. It goes to `Ok` variant in most cases.